src/device/environment.js

```diff
@@ -0,0 +1,110 @@
+createNameSpace("realityEditor.device.environment");
+
+/**
+ * @fileOverview realityEditor.device.environment.js
+ * This provides an extensible location for defining environment variables that are used by
+ * the core application, but may be modified by add-ons to affect conditional behavior in the app.
+ *
+ * For example, an add-on can disable distance fading, if that is important for the add-on behavior,
+ * or it can change which event names the app responds to (mousedown vs touchdown), or it can scale
+ * certain UI constants, such as the link line width, by factors specific to the environment.
+ *
+ * Currently, if multiple add-ons try to set the same variable it will lead to inconsistent results
+ * based on which add-on is loaded first. There are plans to allow add-ons to register their
+ * individual requirements, which the getter functions would resolve at runtime.
+ */
+(function(exports) {
+
+    exports.initService = function() {
+        console.log('Default environment initialized');
+    };
+
+    // initialized with default variables for iPhone environment. add-ons can modify
+    let variables = {
+        // booleans
+        providesOwnUpdateLoop: false,
+        shouldBroadcastUpdateObjectMatrix: false,
+        doWorldObjectsRequireCameraTransform: false,
+        requiresMouseEvents: false,
+        supportsDistanceFading: true,
+        shouldCreateDesktopSocket: false,
+        // numbers
+        lineWidthMultiplier: 1, // 5
+        distanceScaleFactor: 1 // 10
+    };
+
+    // variables can be directly set by add-ons by using the public 'variables' property
+    exports.variables = variables;
```

Comment:
If you want this to be pre-deprecated you could give it a "hidden" name like `exports._variables` or `exports.__variables`

Reply:
I'm not sure if I understand pre-deprecation. Right now I'm using that variable publicly to set its values in the addon, for instance
```
let env = realityEditor.device.environment;
env.variables.providesOwnUpdateLoop = true;
env.variables.requiresMouseEvents = true;
```

Is it bad practice to set it this way, but encourage/require the getter functions for getting the value?

Reply:
If you're using it that way then I agree with the current name. From the comment that I can't find anymore it sounded like you were planning to only ever have setters/getters and never use .variables